exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -69,6 +69,10 @@ type AzureManagedMachinePoolSpec struct {
 	// MaxPods specifies the kubelet --max-pods configuration for the node pool.
 	// +optional
 	MaxPods *int32 `json:"maxPods,omitempty"`
+
+	// OsDiskType specifies the OS disk type for each node in the pool. Allowed values are 'Ephemeral' and 'Managed'.
```

Comment:
lets add the openAPI schema validation for this field
// +kubebuilder:validation:Enum= Ephemeral; Managed